### PR TITLE
(PCP-494) Acceptance test of including non-ASCII character in Puppet manifest

### PR DESCRIPTION
**Pull request by james-stocks**
[skip ci]

### CODE REVIEW

File: acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require 'pxp-agent/test_helper.rb'
+
+ENVIRONMENT_NAME = 'non_ASCII'
+
+test_name 'C98107 - Run puppet with non-ASCII characters in Puppet code' do
+
+  step 'On master, create a new environment that includes a non-ASCII character and will result in Changed' do
+    environmentpath = master.puppet['environmentpath']
```
Comment (er0ck) [inline acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb]: we have a helper for this, and this environment needs a teardown.  
actually we should probably just use mk_tmp_environment_on() or whatever it's called in our libs

File: acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'C98107 - Run puppet with non-ASCII characters in Puppet code' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  env_name = test_file_name = File.basename(__FILE__, '.*')
+  ENVIRONMENT_NAME = mk_tmp_environment(env_name)
```
Comment (er0ck) [inline acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb]: this should be a variable.

File: acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  step 'On master, create a new environment that includes a non-ASCII character and will result in Changed' do
+    environmentpath = master.puppet['environmentpath']
```
Comment (er0ck) [inline acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb]: no longer needed.  `#environmentpath` is a helper in EnvironmentUtils

File: acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    site_manifest = "#{environmentpath}/#{ENVIRONMENT_NAME}/manifests/site.pp"
+    on(master, "cp -r #{environmentpath}/production #{environmentpath}/#{ENVIRONMENT_NAME}")
```
Comment (er0ck) [inline acceptance/tests/pxp-module-puppet/run_puppet_nonASCII.rb]: no longer needed.  the `#mk_tmp_environment_on` links the tmp environment into the environmentpath (but not into production)

**Comment by er0ck** [comment]
let's hold on this until PA ships

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
+  environment_name = mk_tmp_environment(env_name)
 
   step 'On master, create a new environment that will result in Changed' do
-    environmentpath = master.puppet['environmentpath']
-    site_manifest = "#{environmentpath}/#{ENVIRONMENT_NAME}/manifests/site.pp"
-    on(master, "cp -r #{environmentpath}/production #{environmentpath}/#{ENVIRONMENT_NAME}")
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
     create_remote_file(master, site_manifest, <<-SITEPP)
 node default {
   notify {'Notify resources cause a Puppet run to have a \\'changed\\' outcome ☃':}
```
Comment (MikaelSmith) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: This has a snow-man in it, so these are both testing non-ASCII. That's what I meant when I said there was an acceptance test (I added the snowman in my PR fixing the issue).

Comment (james-stocks) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: I didn't notice that, sorry.
This case might warrant having its own test case; so in the event of a test run failing we have 1 test result for a Changed puppet run and a separate test result for a Puppet run with non-ASCII characters.
On the other hand; a single test is more efficient in CI.

We can reject this PR; but we should at least update the test case to advertise that it is testing non-ASCII

Comment (MikaelSmith) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: I'm fine with a single test; we're going to run the test a lot and probably break it pretty rarely. Updating it to advertise the non-ASCII portion makes sense.

**Comment by er0ck** [comment]
This duplicates some testing from changes in the upstream PR.
Subsequent changes will update that test to make it more clear that it's testing unicode puppet output and fixup the test to use the env helper.
